js-functions-arrays/functions-arrays.js

```diff
@@ -0,0 +1,144 @@
+    if (a > b) {
+        return a;
+    }
+    return b;
+}
+myFunctionDeclaration (5,10);
+let resultFunctionDeclaration = myFunctionDeclaration (5,10);
+
+console.log('result of function declaration',resultFunctionDeclaration);
+
+// Task 1.2. Function expression:
+let myFunctionExpression = function(a,b) {
+    if (a > b) {
+       return a;
+    }
+    return b;
+}
+myFunctionExpression (4,0);
+let resultFunctionExpression = myFunctionExpression(4,0);
+
+console.log('result of function expression', resultFunctionExpression);
+
+// 1.3. Arrow function, varian 1:
+let myFunctionArrow = (a, b) => {
+    if (a > b){ 
+        return a;
+    }
+    return b;
+}
+let myFunctionArrowResult = myFunctionArrow (8,3);
+
+console.log('result of arrow function', myFunctionArrowResult);
+
+// 1.3. Arrow function, varian 2:
+let myFunctionArrowBoolean = (a, b) => a > b ? a : b;
+let myFunctionArrowResultSimple = myFunctionArrowBoolean (12,4);
+
+console.log('result of arrow function simple way', myFunctionArrowResultSimple);
+
+// Task 2:
+function addOneParameter(value) {
+    console.log(value);
+    let sumOfValueIndex = value + 1;
+        if (sumOfValueIndex <= 20) {
+            addOneParameter(sumOfValueIndex);
+        }
+    }
+addOneParameter(2);
+
+// Task 3:
+// Variant 1 check with "for"
+let nestedArray = [["apple", "banana", "dog"],["cherry", "cat", "elephant"],["red", "green", "cat"]];
+function checkNestedArrayIndex(nestedArray) {
+    for (let index = 0; index < nestedArray.length; index++) {
+        const element = nestedArray[index];
+        console.log('first array', element);
+        for (let a = 0; a < nestedArray.length; a++) {
+            console.log(nestedArray[index][a]); 
+        }
+    }
+}
+checkNestedArrayIndex(nestedArray);
+
+function checkNestedArrayIncludes(nestedArray) {
+    for (let animal of nestedArray) {
+    for (let cat of animal) {
+        console.log('check cat in array', cat.includes("cat"));
+    }
+    }
+}
+checkNestedArrayIncludes(nestedArray);
+
+// Task 4:
+// assumption made that the first element shall be divided on the last element in the array
+
+const numbers = [1, 2, 10, 3]; 
+function divideArray(numbers) {
+    const result = [];
+    
+    if(numbers.length < 2) {
+        throw new Error('length of Array is too short');
+    }
+
+    try {
+        for (let i = 0; i < numbers.length; i++) {
+            let currentValue = numbers[i];
+            let res;
+            console.log(currentValue);
+            if (typeof currentValue !== 'number' || currentValue === 0) {
+              throw new Error;
+            }
+            if(i === 0) {
+                res = currentValue / numbers[numbers.length - 1];
+            } else {
+                res = currentValue / numbers[i - 1]
+            }
+
+            result.push(res);
+        }
+    }
+    catch (error) { 
+        console.log("error message: this is new error your element in the Array contains not equal to number or zero");
```

Comment:
The error object provides two useful properties: name and message.
pls use: error.message to show error